### PR TITLE
Offline mode: conflict on scheduled posts

### PR DESCRIPTION
**Pull request by pantstamp**
This PR is the companion app for FluxC Post conflict resolution for scheduled posts.
FluxC PR: https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2994

Merge Instructions

Ensure FluxC PR has been merged to trunk
Update this PR with the CI Publish hash from the FluxC Merge into trunk
Remove the Not Ready for Merge label
Merge as normal

## To Test:
Follow the instructions in the FLuxC https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2994

-----

## Regression Notes

1. Potential unintended areas of impact

    - The scheduled post or page is not updated properly

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual tests and existing unit tests

3. What automated tests I added (or what prevented me from doing so)

    - N/A

-----

## PR Submission Checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):

- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/wordpress-mobile/WordPress-Android/milestone/282">24.8</a>. This milestone is due in less than 4 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/src/main/java/org/wordpress/android/ui/posts/PostConflictResolutionFeatureUtils.kt [hunk 1]
```diff
@@ -21,9 +25,42 @@ class PostConflictResolutionFeatureUtils @Inject constructor(
...
+     * publication date, as returned by the backend. This setup can prevent effective post-conflict resolution.
+     * This function introduces an additional field in the payload that is used to override the `lastModified` value.
+     * By setting a past date in `if_not_modified_since`, it ensures that the conflict resolution mechanism can operate
+     * correctly.
+     */
+    private fun setLastModifiedForConflictResolution(payload: RemotePostPayload) {
+        payload.post?.let { post ->
+            payload.lastModifiedForConflictResolution = if (shouldUpdateLastModifiedForConflictResolution(post)) {
+                dateTimeUtilsWrapper.dateStringFromIso8601MinusMillis(
+                    dateTimeUtilsWrapper.currentTimeInIso8601(),
```
Comment (pantstamp) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/PostConflictResolutionFeatureUtils.kt:48]: @zwarm could you please check this line? Your commit didn't work for me so I had to pass the current date minus something.

File: WordPress/src/main/java/org/wordpress/android/ui/posts/PostConflictResolutionFeatureUtils.kt [hunk 2]
```diff
@@ -21,9 +25,42 @@ class PostConflictResolutionFeatureUtils @Inject constructor(
...
+    }
+
+    private fun shouldUpdateLastModifiedForConflictResolution(post: PostModel): Boolean {
+        return post.status == PostStatus.SCHEDULED.toString()
+                && post.remotePostId > 0
+                && post.lastModified == post.dateCreated
+    }
+
+    companion object {
+        const val MILLISECONDS_IN_A_HOUR = 1000*60*60L
```
Comment (pantstamp) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/PostConflictResolutionFeatureUtils.kt:64]: @zwarm I went with a hour just to be sure. If you think we should another value please feel free to suggest it :-)

File: WordPress/src/main/java/org/wordpress/android/ui/posts/PostConflictResolutionFeatureUtils.kt (hunk 1 above)
Comment (zwarm) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/PostConflictResolutionFeatureUtils.kt:48]: @pantstamp very odd indeed, but I tested with current time and all is working as expected

**Review by zwarm — APPROVED** [review]
Nice work @pantstamp . Everything is working as expected

**Pull request merged**